scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -3295,7 +3295,12 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
       case KwExtension() =>
         extensionGroupDecl(mods)
       case KwCase() if dialect.allowEnums && enumCaseAllowed && ahead(token.is[Ident]) =>
-        enumCaseDef(mods)
+        mods.find(_.is[Mod.Open]) match {
```

Comment:
I wonder if we should put it in checks, but that cause invariant exceptions, which are not pretty :thinking: